src/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from "react";
+import {
+  Box,
+  Stack,
+  CheckboxGroup as ChakraCheckboxGroup,
+  useMultiStyleConfig,
+} from "@chakra-ui/react";
+
+import HelperErrorText from "../HelperErrorText/HelperErrorText";
+import generateUUID from "../../helpers/generateUUID";
+import { spacing } from "../../theme/foundations/spacing";
+import { CheckboxGroupLayoutTypes } from "./CheckboxGroupLayoutTypes";
+import Checkbox from "../Checkbox/Checkbox";
+
+export interface CheckboxGroupProps {
+  /** Any child node passed to the component. */
+  children: React.ReactNode;
+  /** Populates the initial value of the input */
+  defaultValue?: string[];
+  /** Optional string to populate the HelperErrorText for error state */
+  invalidText?: string;
+  /** Optional string to populate the HelperErrorText for standard state */
+  helperText?: string;
+  /** ID that other components can cross reference for accessibility purposes */
+  id?: string;
+  /** Adds the 'disabled' prop to the input when true. */
+  isDisabled?: boolean;
+  /** A`dds the 'aria-invalid' attribute to the input and
+   * sets the error state when true. */
+  isInvalid?: boolean;
+  /** Adds the 'required' attribute to the input when true. */
+  isRequired?: boolean;
+  /** The checkbox group label displayed in a `legend` element if `showlabel` is
+   * true, or an "aria-label" if `showLabel` is false. */
+  labelText: string;
+  /** Renders the checkbox buttons in a row or column (default). */
+  layout?: CheckboxGroupLayoutTypes;
+  /** The `name` prop indicates the form group for all the `Checkbox` children. */
+  name: string;
+  /** The action to perform on the `<input>`'s onChange function  */
+  onChange?: (value: string[]) => void;
+  /** Whether or not to display "Required"/"Optional" in the label text. */
+  optReqFlag?: boolean;
+  /** Offers the ability to show the group's legend onscreen or hide it. Refer
+   * to the `labelText` property for more information. */
+  showLabel?: boolean;
+}
+
+const noop = () => {};
+
+/**
+ * Wrapper component to wrap `Checkbox` components. Can be displayed in a
```

Comment:
This is the description that is showing at the top of the story `Docs` screen.  Why use this and not something written right into the story file?

Wherever the final description sits, it should include details similar to what was included in the `Select` description.

```
The `CheckboxGroup` component renders all the necessary wrapping and associated text elements, but the checkbox input elements _need_ to be child `Checkbox` components from the NYPL Design System.`

Reply:
I find it easier to reuse the dev comment description for a component in Storybook using `<Description of={CheckboxGroup} />`. I updated the comment to include the above description.